src/internal/components/pluggableVisualizations/bulletChart/bucketHelper.ts

```diff
@@ -0,0 +1,60 @@
+// (C) 2020 GoodData Corporation
+import * as BucketNames from "../../../../constants/bucketNames";
+import { IBucket, IBucketItem } from "../../../interfaces/Visualization";
+import { METRIC } from "../../../constants/bucket";
+import {
+    getPreferredBucketItems,
+    getAllAttributeItems,
+    limitNumberOfMeasuresInBuckets,
+} from "../../../utils/bucketHelper";
+
+const orderedMeasureBucketsLocalIdentifiers = [
+    BucketNames.MEASURES,
+    BucketNames.SECONDARY_MEASURES,
+    BucketNames.TERTIARY_MEASURES,
+];
+
+const placeMeasures = (buckets: IBucket[]) => {
+    const emptyBucketsLocalIdentifiers: string[] = [];
+    const bucketsWithPlacedMeasure: IBucket[] = [];
+    const measuresToBePlaced: IBucketItem[] = [];
+
+    orderedMeasureBucketsLocalIdentifiers.forEach((bucketLocalIdentifier: string) => {
+        const preferredBucketItems = getPreferredBucketItems(buckets, [bucketLocalIdentifier], [METRIC]);
+
+        if (preferredBucketItems.length === 0) {
+            emptyBucketsLocalIdentifiers.push(bucketLocalIdentifier);
+        } else {
+            bucketsWithPlacedMeasure.push({
+                localIdentifier: bucketLocalIdentifier,
+                items: preferredBucketItems.splice(0, 1),
+            });
+
+            preferredBucketItems.forEach(bucketItem => measuresToBePlaced.push(bucketItem));
+        }
+    });
+
+    emptyBucketsLocalIdentifiers.forEach((bucketLocalIdentifier: string, index) => {
+        bucketsWithPlacedMeasure.push({
+            localIdentifier: bucketLocalIdentifier,
+            items: measuresToBePlaced.slice(index, index + 1),
+        });
+    });
+
+    return orderedMeasureBucketsLocalIdentifiers.map((bucketLocalIdentifier: string) =>
+        bucketsWithPlacedMeasure.find((bucket: IBucket) => bucket.localIdentifier === bucketLocalIdentifier),
+    );
+};
+
+export const transformBucketItems = (buckets: IBucket[]): IBucket[] => {
```

Comment:
I cannot come up with a good name but as it is working with buckets, I would call it `transformBuckets`